# GtkNotebook3: add classes for stack widget
# default way: notebook header.top.bottom.left.right + notebook stack
# updated way: notebook header.top.bottom.left.right + notebook stack.top.bottom.left.right
# https://github.com/GNOME/gtk/blob/3.24.37/gtk/gtknotebook.c
Index: b/gtk/gtknotebook.c
===================================================================
--- a/gtk/gtknotebook.c
+++ b/gtk/gtknotebook.c
@@ -1351,6 +1351,7 @@ gtk_notebook_init
                                                   gtk_notebook_draw_stack,
                                                   NULL,
                                                   NULL);
+  gtk_css_gadget_add_class (priv->stack_gadget, GTK_STYLE_CLASS_TOP);
   gtk_css_gadget_set_state (priv->stack_gadget, gtk_css_node_get_state (widget_node));
   gtk_box_gadget_insert_gadget (GTK_BOX_GADGET (priv->gadget), -1, priv->stack_gadget, TRUE, GTK_ALIGN_FILL);
 
@@ -7074,9 +7075,15 @@ gtk_notebook_update_tab_pos
   for (i = 0; i < G_N_ELEMENTS (tab_pos_names); i++)
     {
       if (tab_pos == i)
+      {
+        gtk_css_gadget_add_class (priv->stack_gadget, tab_pos_names[i]);
         gtk_css_gadget_add_class (priv->header_gadget, tab_pos_names[i]);
+      }
       else
+      {
+        gtk_css_gadget_remove_class (priv->stack_gadget, tab_pos_names[i]);
         gtk_css_gadget_remove_class (priv->header_gadget, tab_pos_names[i]);
+      }
     }
 
   gtk_box_gadget_remove_gadget (GTK_BOX_GADGET (priv->gadget), priv->header_gadget);
